### PR TITLE
Add snackbar to appropriate places

### LINKED ISSUE #399: Add Snackbar to appropriate places
## Description

Add snackbar components to appropriate places to give user a good indication on errors and success

## Possible Implementation

https://mui.com/components/snackbars/

### PR DESCRIPTION
**Pull request by willyyhuang**
Closes #399 

Added notistack which provides snackbar queue functionalities to improve our application's UX.

### CODE REVIEW

**Review by m-triassi — CHANGES_REQUESTED** [review]
You're going to need to get your test coverage above the threshold for this. If your doing unplanned tasks that don't contribute towards completing a feature they have to be well tested.

File: public/translation_fr.json [hunk 1]
```diff
@@ -26,7 +26,9 @@
         "import-note": "Note: Les personnages importés auront des champs vide pour les objets et sessions, merci de les saisir manuellement.",
         "enter-details": "Ajouter des détails",
         "privacy-details": "Confidentialité",
-        "import-help-text-3": "Entrer votre lien de la fiche sur D&D Beyond."
+        "import-help-text-3": "Entrer votre lien de la fiche sur D&D Beyond.",
+        "create-success-message": "Personnage crée",
+        "edit-success-message": "Personnage mise à jour"
```
Comment (pascwhale) [inline public/translation_fr.json:31]: I feel like this should say `Personnage mis à jour` instead.

File: public/translation_fr.json [hunk 2]
```diff
@@ -73,7 +75,9 @@
         "end-date-and-time": "Date et heure de la fin",
         "fill-out-fields-dm": "Saisissez les champs suivants avec la session du MJ.",
         "fill-out-fields-entry": "Saisissez les champs suivants de la session.",
-        "no-adventure-warning": "Cette session importée n'a pas d'aventure associée, veuillez l'ajouter manuellement."
+        "no-adventure-warning": "Cette session importée n'a pas d'aventure associée, veuillez l'ajouter manuellement.",
+        "create-success-message": "Session crée",
```
Comment (pascwhale) [inline public/translation_fr.json:79]: This one should be `Session créée`.

**Comment by willyyhuang** [comment]
> You're going to need to get your test coverage above the threshold for this. If your doing unplanned tasks that don't contribute towards completing a feature they have to be well tested.

As much as I want to increase the coverage, I have no way of mocking an API response to trigger a change in inertia's `useForm`'s `wasSuccessful` flag after a `post` or `put` request, I have tried manually mocking the function but have no luck at all, I believe the 3 lines added to old code can not be tested unless we can find a way of mocking actual backend response in front-end unit test.

This task is also part of the planned task created last sprint to improve UX, it also helps in features as we are using `Snackbar` in places.

**Review by pascwhale — COMMENTED** [review]
I had a question here. How is it that you're able to remove the `ThemeProvider`, `getFontTheme` and `useUser` import statements everywhere?

I also noticed that the success snackbar was not the correct green according to our color palette. Is there any way you can change that?

File: public/translation_fr.json (hunk 1 above)
Comment (willyyhuang) [reply public/translation_fr.json]: done

File: public/translation_fr.json (hunk 2 above)
Comment (willyyhuang) [reply public/translation_fr.json]: done

**Comment by willyyhuang** [comment]
> I had a question here. How is it that you're able to remove the `ThemeProvider`, `getFontTheme` and `useUser` import statements everywhere?
> 
> I also noticed that the success snackbar was not the correct green according to our color palette. Is there any way you can change that?

Addressing the first question, `ThemeProvider` wrapping is moved to `ApplicationLayout`, which ensures that the content all have the same theme while the header all have the regular theme, `useUser` imports are still there if `language` is not needed.

2nd question, yes, I can change to the correct green, I will take a look tomorrow

**Comment by willyyhuang** [comment]
> I also noticed that the success snackbar was not the correct green according to our color palette. Is there any way you can change that?

It is done @pascwhale

**Review by m-triassi — APPROVED** [review]
Alright, works well enough for me

**Pull request merged**